src/main/kotlin/org/ethereum/lists/tokens/TokenChecker.kt

```diff
@@ -37,20 +41,27 @@ val onChainIgnore by lazy {
     File("onChainIgnore.lst").readText().split("\n")
 }
 
-suspend fun checkTokenFile(file: File, onChainCheck: Boolean = false) {
+suspend fun checkTokenFile(file: File, onChainCheck: Boolean = false, chainId: ChainId? = null) {
     val handle = file.name.removeSuffix(".json")
     if (onChainCheck && (notToProcessFiles.contains(handle) || onChainIgnore.contains(handle))) {
         return
     }
 
     val jsonObject = Klaxon().parseJsonObject(file.reader())
     val address = Address(jsonObject["address"] as String)
+    val address_eip1191 = if (jsonObject["address_eip1191"] != null) Address(jsonObject["address_eip1191"] as String) else null
+
     when {
         !address.isValid() -> throw InvalidAddress(address)
 
-        !address.hasValidERC55Checksum()
+        address_eip1191 != null && !(address_eip1191.isValid()) -> throw InvalidAddress(address)
+
+        (!address.hasValidERC55Checksum() && address_eip1191 == null)
```

Comment:
This should also throw an error when the address_eip1191 is non null. We need to make sure that the address field is always valid

Reply:
I did because we need the name of the file has the checksum following the eip1191 when it is appropriate. This allow us to have a better integration with Ledger and Trezor wallets for the different tokens that are in RSK.

Maybe something that we can do is this:
- if `address_eip1191` is empty or null, it will be validated using erc55 rules the `address` field. (as it is working today)
- if `address_eip1191` has a value, it will be validate with erc1191 rules and also the `address` field. So basically the two fields has to be the same.

Another option is to modify the file name validation and allow to check the file name by `address_eip1191` if it is appropriate. 

Let me know what are your thought about this. Thanks!

Reply:
unfortunately this will not work - we cannot break the guarantees that the address field brings.
To have better TREZOR integration you need to change it on the TREZOR side - meaning there the check if the address_eip1191 field is there and if it is use it instead of the address field.

Reply:
Ok, let's do what you have suggested. I will do the modifications and update the PR.